slick/src/sphinx/code/Connection.scala

```diff
@@ -103,6 +105,29 @@ object Connection extends App {
       val f: Future[Unit] = db.run(a)
       //#transaction
       Await.result(f, Duration.Inf)
+    };{
+      //#rollback
+      val insertCoffeesAction = coffees ++= Seq (
+        ("Cold_Drip", new SerialBlob(Array[Byte](101))),
+        ("Dutch_Coffee", new SerialBlob(Array[Byte](49)))
+      )
+
+      val countAction = coffees.length.result.map { coffeesCount: Int =>
+        if(coffeesCount < 5000) {
+          DBIO.failed(new Exception())
+        }
+      }
```

Comment:
Yep, that's what I meant. Isn't your original code wrong? In other words, if you replace ```Await.result(f, Duration.Inf)``` with ```assert(Await.result(f, Duration.Inf) === "Now that is a lot of coffee variety.")``` (probably a good idea) you get an exception, but with the above new code it succeeds?